enterprise_access/apps/api/v1/views.py

```diff
@@ -816,11 +820,23 @@ def redeem(self, request, *args, **kwargs):
 
         learner_id = serializer.data['learner_id']
         content_key = serializer.data['content_key']
+        try:
+            lock_acquired = acquire_subsidy_lock(
+                policy.subsidy_uuid,
+                django_cache_timeout=SUBSIDY_LOCK_TIMEOUT_SECONDS,
+            )
+            if not lock_acquired:
+                return Response(
+                    data='Enrollment currently locked for this subsidy.',
+                    status=status.HTTP_423_LOCKED,
+                )
+            if policy.can_redeem(learner_id, content_key):
```

Comment:
@iloveagent57 [copying your code](https://github.com/openedx/license-manager/blob/f901ea018e8525170e887ff46abed03cd257cebe/license_manager/apps/api/v1/views.py#L742) from license manager. Should there be a lock_id that is sent across to the subsidy api so that it can verify that the calling client is indeed the one holding the lock?

Reply:
Also, the lock_id might need to be sent in such a manner that the receiving api can check that it is not spoofed

Reply:
The lock_id is essentially the cross-service and unique transaction identifier

Reply:
should we retry to acquire the lock if we fail to acquire it ... what should be the retry algorithm ... exponential backoff?